packages/shared/lib/core/wallet/utils/generateActivity/generateSingleBasicActivity.ts

```diff
@@ -43,15 +43,13 @@ export async function generateSingleBasicActivity(
 
     const sendingInfo = getSendingInformation(processedTransaction, output, account)
     const asyncData = await getAsyncDataFromOutput(output, outputId, claimingData, account)
-
     const { parsedLayer2Metadata, destinationNetwork } = getLayer2ActivityInformation(metadata, sendingInfo)
     const layer2Allowance = Number(parsedLayer2Metadata?.baseTokens ?? '0')
-    const gasBudget = Number(parsedLayer2Metadata?.gasBudget ?? '0')
     const gasFee = layer2Allowance > 0 ? amount - layer2Allowance : 0
 
     let { storageDeposit, giftedStorageDeposit } = await getStorageDepositFromOutput(account, output)
     giftedStorageDeposit = action === ActivityAction.Burn ? 0 : giftedStorageDeposit
-    giftedStorageDeposit = gasBudget === 0 ? giftedStorageDeposit : 0
+    giftedStorageDeposit = gasFee === 0 ? giftedStorageDeposit : 0
```

Comment:
I am not very sure about this change, I think it is necessary to maintain the `budget` here